src/python/gudhi/alpha_complex.pyx

```diff
@@ -31,6 +31,10 @@ cdef extern from "Alpha_complex_interface.h" namespace "Gudhi":
         Alpha_complex_interface(vector[vector[double]] points, vector[double] weights, bool fast_version, bool exact_version) nogil except +
         vector[double] get_point(int vertex) nogil except +
         void create_simplex_tree(Simplex_tree_interface_full_featured* simplex_tree, double max_alpha_square, bool default_filtration_value) nogil except +
+        @staticmethod
+        void set_float_relative_precision(double precision) nogil
```

Comment:
IIRC for super cheap functions, it is better not to write `nogil`, because releasing and taking the lock again costs more than the function itself. I could be wrong though.

Reply:
`nogil` removed on f4b543b

Reply:
Actually, I was remembering wrong. Writing `nogil` on those declarations only means it is allowed to call them without the GIL, but it doesn't automatically release the GIL, so you were right to write `nogil`. Could you add it back please? Sorry...

Reply:
I double checked, and it seems releasing the GIL is just like setting a variable and should not take a lot of time. I think we could use `nogil` for all.
Reverted on 25172af